test/unit/spec/v2/view-builder/views/IdentifierView_spec.js

```diff
@@ -49,4 +54,23 @@ describe('v2/view-builder/views/IdentifierView', function() {
     expect(testContext.view.$el.find('.siw-main-footer .js-forgot-password').length).toEqual(1);
     expect(testContext.view.$el.find('.links-primary .js-forgot-password').length).toEqual(0);
   });
+
+  it('view renders IDP buttons correctly with idpDisplay property', function() {
+    jest.spyOn(AppState.prototype, 'hasRemediationObject').mockReturnValue(true);
+    jest.spyOn(AppState.prototype, 'getActionByPath').mockReturnValue(true);
+    jest.spyOn(AppState.prototype, 'isIdentifierOnlyView').mockReturnValue(false);
+    idpDisplay = 'PRIMARY';
+    testContext.init();
+
+    // The idp buttons should be rendered above the main form.
+    expect(testContext.view.$el.find('.o-form-fieldset-container .sign-in-with-idp').length).toEqual(1);
+    expect(testContext.view.$el.find('.o-form-button-bar .sign-in-with-idp').length).toEqual(0);
+
+    idpDisplay = 'SECONDARY';
```

Comment:
what if `idpDisplay` is undefined/null/or something garbage.

Reply:
I will add a case for this.